eth/eventservices/rewardservice.go

```diff
@@ -120,22 +120,18 @@ func (s *RewardService) tryReward() error {
 			if err != nil {
 				return err
 			}
+			s.pendingTx = tx
```

Comment:
If we successfully submit a replacement tx on L108, then that tx will be the latest pending tx so any additional gas price bumps would need to be calculated based off of the gas price for that tx. If we only set the pending tx here then I think the following is possible:

1. Submit original tx with nonce N with gas price X
2. Original tx with nonce N is not confirmed within the timeout
3. Submit replacement tx 1 with nonce N with gas price 1.1X
4. Replacement tx 1 with nonce N is not confirmed within the timeout
5. Submit replacement tx 2 with nonce N with gas price 1.1X
    - Note that the gas price is 1.1X since it is calculated off of the gas price for the original tx instead of the gas price for replacement tx 1 (which would be 1.21X)

In order to avoid the above scenario, we should be able to set the pending tx both after a successful `Reward()` call and after a successful `ReplaceTransaction()` call i.e.

```golang
if s.pendingTx != nil {
    tx, err = s.client.ReplaceTransaction(s.pendingTx, "reward", nil)
    ...
} else {
    tx, err = s.client.Reward()
    ...
}

s.pendingTx = tx
```

Reply:
Yeah acknowledged, although I don't think speeding up more than once is the right strategy here though (cancellations might be cheaper and better in that case).

Given urgency though let's roll with what you propose and consider additional strategies down the line. WDYT?

Reply:
fixed in b06500f1